src/transformers/trainer_utils.py

```diff
@@ -206,6 +206,18 @@ class TrainOutput(NamedTuple):
 _re_checkpoint = re.compile(r"^" + PREFIX_CHECKPOINT_DIR + r"\-(\d+)$")
 
 
+def get_checkpoints(folder, reverse=False):
+    content = os.listdir(folder)
+    checkpoints = [
+        path
+        for path in content
+        if _re_checkpoint.search(path) is not None and os.path.isdir(os.path.join(folder, path))
+    ]
+    return map(
+        lambda x: os.path.join(folder, x),
+        sorted(checkpoints, key=lambda x: int(_re_checkpoint.search(x).groups()[0]), reverse=reverse)
+    )
```

Comment:
We have to return a list here, otherwise the checks in trainer.py e.g. `if not checkpoints` and `len(checkpoints) > 0` won't work 

```suggestion
    return list(map(
        lambda x: os.path.join(folder, x),
        sorted(checkpoints, key=lambda x: int(_re_checkpoint.search(x).groups()[0]), reverse=reverse)
    ))
```